nucypher/cli/commands/bob.py

```diff
@@ -286,13 +288,15 @@ def public_keys(general_config, character_options, config_file):
 @option_config_file
 @BobInterface.connect_cli('retrieve')
 @group_general_config
+@click.option('--ipfs', help="Download an encrypted message from IPFS at the specified gateway URI")
```

Comment:
Should we generalize this option to some storage gateway uri (i.e. allow for extension to other storage technologies eg. filecoin, swarm etc.) with `ipfs://` as a specific scheme that is implemented?

Similar comment for other cases of `--ipfs` option.

Reply:
Interesting idea, perhaps this can be combined with the `--message-kit`?

Reply:
Interested to better understand what you mean, but I would caution that to remain storage agnostic, it may be best not to mix the message kit with the underlying storage medium.

Reply:
You are close to my general sentiment on this PR...  Mostly advancing the requirements placed by hackathon participants.

Reply:
What do you think is the resolution here?

Reply:
Phew - that's asking for alot of vision!

Reply:
Haha - I just mean, are we keeping `--ipfs` and then doing a follow-up PR later for something quick, or should we just make something general (eg. `--getway-uri`) instead and have ipfs as the only thing supported for now...

Reply:
This is so minimal it even deserves the proverbial `--yolo` flag :sweat_smile:

Reply:
#2108